### PR TITLE
[MDS-5125] character limit on mine alerts

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[[MDS-5121](https://bcmines.atlassian.net/browse/MDS-5121)]

- Display the number of characters that are already input (as a fraction of the total number allowed) - to match the design.

 - The input shouldn’t allow more characters than what is allowed.
 
image

### CODE REVIEW

File: services/core-web/src/components/common/RenderAutoSizeField.js [hunk 1]
```diff
@@ -16,37 +16,60 @@ const propTypes = {
...
+      }
+    >
+      <Input.TextArea
+        disabled={props.disabled}
+        id={props.id}
+        {...props.input}
+        autoSize={{ minRows: props.minRows }}
+        placeholder={props.placeholder}
+        onChange={handleTextAreaChange}
+        value={value}
```
Comment (matbusby-fw) [inline services/core-web/src/components/common/RenderAutoSizeField.js:63]: Cool, this should definitely work.  I know AntD does have a [`showCount`](https://ant.design/components/input#components-input-demo-show-count) prop for the `Input.TextArea` that might make it a little bit simpler.  

I haven't played with it myself though so maybe that doesn't provide quite the behaviour we're looking for.  Did you look into that at all?

Comment (henryoforeh-dev) [reply services/core-web/src/components/common/RenderAutoSizeField.js:63]: Great, I will have a look at it and refactor. Thanks.

File: services/core-web/src/components/common/RenderAutoSizeField.js [hunk 2]
```diff
@@ -16,37 +16,60 @@ const propTypes = {
...
   minRows: PropTypes.number,
+  maximumCharacters: PropTypes.number,
 };
 
 const defaultProps = {
   placeholder: "",
   label: "",
   disabled: false,
   minRows: 3,
+  maximumCharacters: 300,
```
Comment (taraepp) [inline services/core-web/src/components/common/RenderAutoSizeField.js]: This component is used in quite a few places, I'm not 100% sure we want to enable character limiting by default. Have we done any investigation into that?

Comment (henryoforeh-dev) [reply services/core-web/src/components/common/RenderAutoSizeField.js]: Nice catch, I will refactor.

**Review by taraepp — APPROVED** [review]
LGTM

**Pull request merged**